lib/sles4sap_publiccloud.pm

```diff
@@ -110,6 +112,76 @@ sub get_promoted_hostname {
     return join("", @master);
 }
 
+=head2 sles4sap_cleanup
+
+    Clean up Network peering and qesap deployment
+
+=cut
+
+sub sles4sap_cleanup {
+    my ($self, %args) = @_;
+
+    # If there's an open ssh connection to the VMs, return to host console first
+    select_host_console(force => 1);
+    record_info(
+        'Cleanup',
+        join(' ',
+            'cleanup_called:',
+            $args{cleanup_called} // 'undefined',
+            'network_peering_present:',
+            $args{network_peering_present} // 'undefined',
+            'ansible_present:',
+            $args{ansible_present} // 'undefined')
+    );
+
+    qesap_upload_logs();
+    if ($args{network_peering_present}) {
+        delete_network_peering();
+    }
+
+    # Do not run destroy if already executed
+    return 0 if ($args{cleanup_called});
```

Comment:
from now on this is never reachable condition

Reply:
True, when `sles4sap_cleanup` is called as a `post_fail_hook` in `basetest.pm`, this wont be reached. But since we use this in other parts of our code as well, and we are also using this variable, it could be reached in other cases (in specific cases when it's called as part of `qesap_cleanup` module, for example).

Reply:
ok